src/Internal/DependencyManagement/ExtendedContainer.php

```diff
@@ -149,4 +149,14 @@ protected function get_class_from_concrete( $concrete ) {
 	protected function is_class_allowed( string $class_name ): bool {
 		return StringUtil::starts_with( $class_name, $this->woocommerce_namespace, false ) || in_array( $class_name, $this->registration_whitelist, true );
 	}
+
+	/**
+	 * Check if a class name corresponds to an anonoymous class.
+	 *
+	 * @param string $class_name The class name to check.
+	 * @return bool True if the name correspondos to an anynymous class.
```

Comment:
Typo on `correspondos` should be `corresponds`. Typo on `anynymous` should be `anonymous`